src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -249,6 +264,10 @@ public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryComplete
                     if (this.discoveryCompleted != null)
                     {
                         this.discoveryCompleted.Set();
+
+                        //Send the Discovery Complete Raw Message
+                        this.discoveryCompleteTask.Wait();
```

Comment:
Should the order be `rawMessage` is send first, `discoveryCompleted` is set?

Reply:
It is waiting for this event to be set before it can send message, that's why the order